services/core/FileWatchPublisher/filewatchpublisher/agent.py

```diff
@@ -0,0 +1,159 @@
+# nor any of their employees, nor any jurisdiction or organization that
+# has cooperated in the development of these materials, makes any
+# warranty, express or implied, or assumes any legal liability or
+# responsibility for the accuracy, completeness, or usefulness or any
+# information, apparatus, product, software, or process disclosed, or
+# represents that its use would not infringe privately owned rights.
+#
+# Reference herein to any specific commercial product, process, or
+# service by trade name, trademark, manufacturer, or otherwise does not
+# necessarily constitute or imply its endorsement, recommendation, or
+# favoring by the United States Government or any agency thereof, or
+# Battelle Memorial Institute. The views and opinions of authors
+# expressed herein do not necessarily state or reflect those of the
+# United States Government or any agency thereof.
+#
+# PACIFIC NORTHWEST NATIONAL LABORATORY
+# operated by BATTELLE for the UNITED STATES DEPARTMENT OF ENERGY
+# under Contract DE-AC05-76RL01830
+
+#}}}
+
+import logging
+import sys
+
+from datetime import datetime
+from volttron.platform.agent.utils import watch_file_with_fullpath
+from volttron.platform.vip.agent import Agent, RPC, Core
+from volttron.platform.agent import utils
+
+
+utils.setup_logging()
+_log = logging.getLogger(__name__)
+__version__ = '3.6'
+
+
+def file_watch_publisher(config_path, **kwargs):
+    """Load the FileWatchPublisher agent configuration and returns and instance
+    of the agent created using that configuration.
+
+    :param config_path: Path to a configuration file.
+
+    :type config_path: str
+    :returns: FileWatchPublisher agent instance
+    :rtype: FileWatchPublisher agent
+    """
+    config = utils.load_config(config_path)
+    return FileWatchPublisher(config, **kwargs)
+
+
+class FileWatchPublisher(Agent):
+    """Monitors files for changes and publishes added lines
+        on corresponding topics.
+
+    :param config: Configuration dict
+    :type config: dict
+
+    Example configuration:
+
+    .. code-block:: python
+
+        {
+	        "publish_file": [
+                {
+                    "file": "/var/log/syslog",
+                    "topic": "platforms/syslog",
+                },
+                {
+                    "file": "volttron.log",
+                    "topic": "platforms/log",
+                }
+	        ]
+        }
+    """
+    def __init__(self, config, **kwargs):
+        super(FileWatchPublisher, self).__init__(**kwargs)
+        self.config = config
+        self.file_topic = {}
+        self.file_end_position = {}
+        for item in self.config:
+            file =  item["file"]
+            self.file_topic[file] = item["topic"]
+            with open(file, 'r') as f:
+                self.file_end_position[file] = self.get_end_position(f)
+
+    @Core.receiver('onstart')
+    def starting(self, sender, **kwargs):
+        _log.info("Starting FileWatchPublisher agent")
```

Comment:
instead of embedding the class name you can use self.__class__.__name__ or self.__name__.  That way if we change it later we won't have to change it here.  Note the self.__name__ is only relevant if you set the instance on line 84 to the __name__ of the class.